tests/components/mqtt/test_init.py

```diff
@@ -3805,10 +3805,12 @@ def _check_entities() -> int:
     async_fire_mqtt_message(hass, "test-topic_manual1", "manual1_intial")
     async_fire_mqtt_message(hass, "test-topic_manual3", "manual3_intial")
 
-    assert (state := hass.states.get("sensor.test_manual1")) is not None
```

Comment:
Out of curiosity, why is this changed?

Reply:
There is an open bug in pytest assertion rewriter. See https://github.com/pytest-dev/pytest/issues/11239

Tbh I'm not sure if we should update. The intention behind https://github.com/home-assistant/core/pull/96995/commits/8579d5f3e3e2355a604ba2e3ef068f3ee89a3a40 was more to see if CI would pass (it should 🤞🏻). `7.4.0` fixed a lot of deprecation warnings which would appear in Python 3.12. What do you think, should we update now even if `:=` in asserts don't work?

Reply:
Are these the only places where assertions are used using the walrus? I don't think they're that many tbh

Without the walrus the test is better to read imo

Reply:
> Are these the only places where assertions are used using the walrus? I don't think they're that many tbh

AFAIC yes. At least those two files are the only ones with errors during the last CI run.
https://github.com/home-assistant/core/actions/runs/6060843262

What's difficult IMO is that the test case using `:=` is normally fine. It's actually an unrelated test case in the same test file that breaks which just happens to use the same name (`state` in this case) in an assert. This should really be fixed in pytest.

> Without the walrus the test is better to read imo

It depends. I'm probably +0 on this. Somethings can I see the appeal of using assignment expressions.

Reply:
Opened a PR upstream to fix it: https://github.com/pytest-dev/pytest/pull/11414